src/Action.js

```diff
@@ -0,0 +1,28 @@
+export default class Action {
+	constructor (promise) {
+		this.promise = promise
+	}
+
+	// default onFulfilled action
+	/* istanbul ignore next */
+	fulfilled (p) {
+		this.promise._become(p)
+	}
+
+	// default onRejected action
+	rejected (p) {
+		this.promise._become(p)
+		return false
+	}
+
+	tryCall (f, x) {
```

Comment:
`this.handle` is expected to never throw, I really wanted to wrap only the user function in the try-block.
Unfortunately JS doesn't have `try`-`except`-`else` like Python :-/
Under the *assumption* that `this.handle` does not throw they should be equivalent indeed, but I found it cleaner to be explicit. Not sure about performance implications.